core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyValueCommand.java

```diff
@@ -28,25 +28,25 @@
 import org.infinispan.util.logging.Log;
 import org.infinispan.util.logging.LogFactory;
 
-public final class ReadWriteKeyValueCommand<K, V, R> extends AbstractWriteKeyCommand<K, V> {
+public final class ReadWriteKeyValueCommand<K, V, T, R> extends AbstractWriteKeyCommand<K, V> {
    private static final Log log = LogFactory.getLog(ReadWriteKeyValueCommand.class);
 
    public static final byte COMMAND_ID = 51;
 
-   private Object value;
-   private BiFunction<V, ReadWriteEntryView<K, V>, R> f;
+   private Object argument;
+   private BiFunction<T, ReadWriteEntryView<K, V>, R> f;
    private Object prevValue;
    private Metadata prevMetadata;
 
-   public ReadWriteKeyValueCommand(Object key, Object value, BiFunction<V, ReadWriteEntryView<K, V>, R> f,
+   public ReadWriteKeyValueCommand(Object key, Object argument, BiFunction<T, ReadWriteEntryView<K, V>, R> f,
```

Comment:
why are you renaming value to argument ?

Reply:
Because once we've unlinked type of value-in-cache from the argument we're passing, it makes no sense to call this 'value'. It's rather 'delta', or generally anything.